ts/features/idpay/barcode/screens/IdPayBarcodeResultScreen.tsx

```diff
@@ -0,0 +1,232 @@
+import {
+  Body,
+  GradientScrollView,
+  H2,
+  H3,
+  HeaderSecondLevel,
+  IOColors,
+  IOVisualCostants,
+  LabelSmall,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { RouteProp, useNavigation, useRoute } from "@react-navigation/native";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import * as React from "react";
+import { SafeAreaView, StyleSheet, View } from "react-native";
+import Barcode from "react-native-barcode-builder";
+import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
+import { OperationResultScreenContent } from "../../../../components/screens/OperationResultScreenContent";
+import { LoadingIndicator } from "../../../../components/ui/LoadingIndicator";
+import I18n from "../../../../i18n";
+import { useIOSelector } from "../../../../store/hooks";
+import { ProgressBar } from "../../../bonus/bpd/screens/details/components/summary/base/ProgressBar";
+import { IDPayDetailsRoutes } from "../../details/navigation";
+import { IdPayBarcodeParamsList } from "../navigation/params";
+import { idPayBarcodeByInitiativeIdSelector } from "../store";
+import { calculateIdPayBarcodeSecondsToExpire } from "../utils";
+
+// -------------------- types --------------------
+
+type IdPayBarcodeResultRouteParams = {
+  initiativeId: string;
+};
+type IdPayBarcodeResultRouteProps = RouteProp<
+  IdPayBarcodeParamsList,
+  "IDPAY_BARCODE_RESULT"
+>;
+
+// -------------------- main component --------------------
+
+const IdPayBarcodeResultScreen = () => {
+  const route = useRoute<IdPayBarcodeResultRouteProps>();
+  const { initiativeId } = route.params;
+  const navigation = useNavigation();
+  const barcodePot = useIOSelector(state =>
+    idPayBarcodeByInitiativeIdSelector(state)(initiativeId)
+  );
+
+  const navigateToInitiativeDetails = () =>
+    navigation.navigate(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
+      route: IDPayDetailsRoutes.IDPAY_DETAILS_MONITORING,
+      routeParams: { initiativeId }
+    });
+
+  if (pot.isLoading(barcodePot)) {
+    return <LoadingScreen />;
+  }
+  return pipe(
+    barcodePot,
+    pot.toOption,
+    O.fold(
+      () => <ErrorContent onCtaPress={navigateToInitiativeDetails} />,
```

Comment:
Why making another component that does the same of a ready to use component? :)

```suggestion
      () => <OperationResultScreenContent
    title={I18n.t("idpay.barCode.resultScreen.error.generic.body")}
    action={{
      label: I18n.t("global.buttons.close"),
      accessibilityLabel: I18n.t("global.buttons.close"),
      onPress: navigateToInitiativeDetails
    }}
    pictogram="umbrellaNew"
  />,
```

Reply:
this was done in scope of future complete error handling, so that the update to a more granular handling would be easier to implement; do you think it's too much foreshadowing?

Reply:
Let's keep things simple for now 😀

Reply:
as you wish, added in the next commit 👍